### PR TITLE
resources map to jaeger process tags

### PR DESCRIPTION
**Pull request by danrusei**
#205 
I hope my understanding about what is requested here is correct, otherwise I'll modify based on the feedback.

### CODE REVIEW

File: opentelemetry-jaeger/src/lib.rs [hunk 1]
```diff
@@ -211,8 +211,23 @@ impl trace::SpanExporter for Exporter {
     fn export(&self, batch: Vec<Arc<trace::SpanData>>) -> trace::ExportResult {
         match self.uploader.lock() {
             Ok(mut uploader) => {
-                let jaeger_spans = batch.into_iter().map(Into::into).collect();
-                uploader.upload(jaeger::Batch::new(self.process.clone(), jaeger_spans))
+                let mut jaeger_spans: Vec<jaeger::Span> = Vec::new();
+                let mut process_tags: Vec<jaeger::Tag> = Vec::new();
+
+                for span in batch.iter() {
```
Comment (TommyCpp) [inline opentelemetry-jaeger/src/lib.rs:217]: Could just use the `batch.into_iter()` and switching the following two line like 
```rust
process_tags.extend(build_process_tags(&span));
jaeger_spans.push(span.into());
```

File: opentelemetry-jaeger/src/lib.rs [hunk 2]
```diff
@@ -431,7 +446,7 @@ impl Into<jaeger::Log> for api::Event {
     }
 }
 
-impl Into<jaeger::Span> for Arc<trace::SpanData> {
+impl Into<jaeger::Span> for &Arc<trace::SpanData> {
```
Comment (TommyCpp) [inline opentelemetry-jaeger/src/lib.rs:449]: Probably no need to change this if we could just use `batch.into_iter()` above.

**Review by TommyCpp — CHANGES_REQUESTED** [review]
Good start, thanks! have some suggestions

File: opentelemetry-jaeger/src/lib.rs [hunk 3]
```diff
@@ -211,8 +211,23 @@ impl trace::SpanExporter for Exporter {
     fn export(&self, batch: Vec<Arc<trace::SpanData>>) -> trace::ExportResult {
         match self.uploader.lock() {
             Ok(mut uploader) => {
-                let jaeger_spans = batch.into_iter().map(Into::into).collect();
-                uploader.upload(jaeger::Batch::new(self.process.clone(), jaeger_spans))
+                let mut jaeger_spans: Vec<jaeger::Span> = Vec::new();
```
[the remark this answers is not in the document]
Comment (danrusei) [reply opentelemetry-jaeger/src/lib.rs:214]: Thank you, I'm updating the PR with the suggested changes. But do we know the capacity of the process_tags ?

[the remark this answers is not in the document]
Comment (TommyCpp) [reply opentelemetry-jaeger/src/lib.rs:214]: Ha sorry I was looking at the wrong variable. We don't know the capacity of the process tags.

**Review by TommyCpp — APPROVED** [review]
Sorry for the late reply. LGTM. Thanks!

File: opentelemetry-jaeger/src/lib.rs [hunk 4]
```diff
@@ -211,8 +211,23 @@ impl trace::SpanExporter for Exporter {
     fn export(&self, batch: Vec<Arc<trace::SpanData>>) -> trace::ExportResult {
         match self.uploader.lock() {
             Ok(mut uploader) => {
-                let jaeger_spans = batch.into_iter().map(Into::into).collect();
-                uploader.upload(jaeger::Batch::new(self.process.clone(), jaeger_spans))
+                let mut jaeger_spans: Vec<jaeger::Span> = Vec::with_capacity(batch.len());
+                let mut process_tags: Vec<jaeger::Tag> = Vec::new();
+
+                for span in batch.into_iter() {
+                    process_tags.extend(build_process_tags(&span));
```
Comment (jtescher) [inline opentelemetry-jaeger/src/lib.rs:218]: Will this duplicate these tags for every span?

**Comment by jtescher** [comment]
@danrusei generally looks good, just want to make sure this won't add duplicate process tags when exporting batches of spans

**Comment by danrusei** [comment]
> @danrusei generally looks good, just want to make sure this won't add duplicate process tags when exporting batches of spans

thank you, would this work ?

    for tag in build_process_tags(&span) {
                        if !process_tags.contains(&tag) {
                            process_tags.push(tag)
                        }
                    }

**Comment by jtescher** [comment]
@danrusei That should work but it may do a lot of work checking each time. The resource _should_ be the same for all spans, so you could do something like `for (idx, tag) in tags.enumerate() { if idx == 0 { add tags} ... }` to only add them the first time?

**Comment by jtescher** [comment]
Looks good @danrusei, just have to rebase and make sure tests pass

**Review by jtescher — APPROVED** [review]
Looks good thanks @danrusei

**Pull request merged**